bin/wasm-node/javascript/src/index.js

```diff
@@ -230,9 +230,17 @@ export function start(config) {
       return chainAddedPromise;
     },
     terminate: () => {
-      worker.terminate();
-      if (!workerError)
-        workerError = new AlreadyDestroyedError();
+      clearTimeout(livenessTimeout)
+
+      // On NodeJS `worker.terminate()` returns a Promise, while on the DOM
+      // it returns `undefined`. In order to have a consistent API we will
+      // always return a Promise
+      const terminationPromise = worker.terminate() || Promise.resolve()
```

Comment:
Could you do this in `compat-node.js` and `compat-browser.js` instead?
Something like `export const workerTerminate = ...` similar to what there already is.

Reply:
mmmm, sure let me give that a shot, that makes sense!

Reply:
done!